url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/Receiver_multicast.java

```diff
@@ -0,0 +1,532 @@
+	    StringBuilder s = new StringBuilder();
+	    s.append("Usage: \n");
+	    s.append("    java -jar Receiver.jar pathToFile fileSize overhead port\n");
+	    s.append("\n        - pathToFile  : path to store the received file.");
+	    s.append("\n        - overhead    : the necessary symbol overhead needed for decoding (usually between 0 and 2).");
+	    s.append("\n        - destIP      : the IP address for the multicast.");
+	    s.append("\n        - portNumber  : the port the receiver will be listening on.");
+	    s.append("\n        - canalloss   : the loss due to the canal.");
+	    s.append("\n        - test   : Para de test.");
+	    s.append("\n        - redondance  : Para de test.");
+	    System.out.println(s.toString());
+	    System.exit(1);
+	}
+	System.out.println(Utils.IntegerSize);
+	Boolean historyInLine = Boolean.valueOf(args[7]);
+	double redondance = Double.valueOf(args[6]);
+	
+	String fileName = args[0];
+	
+	// check fileSize
+	long fileSize = 0;
+	
+	// check overhead
+	int overhead = -1;
+	
+	try {
+	    overhead = Integer.valueOf(args[1]);
+	} catch (NumberFormatException e) {
+	    System.err.println("Invalid overhead value. (must be a positive integer)");
+	    System.exit(-1);
+	}
+	
+	if ( overhead < 0 ) {
+	    System.err.println("Invalid overhead value. (must be a positive integer)");
+	    System.exit(-1);
+	}
+	
+	// get IP and transform to InetAddress
+	InetAddress sendIP = null;
+	
+	try {
+	    sendIP = InetAddress.getByName(args[2]);
+	} catch (UnknownHostException e2) {
+	    e2.printStackTrace();
+	    System.err.println("invalid IP");
+	    System.exit(1);
+	}
+	
+	// check source port
+	int srcPort = -1;
+	
+	try {
+	    srcPort = Integer.valueOf(args[3]);
+	} catch (NumberFormatException e) {
+	    System.err.println("Invalid port. (must be above 1024)");
+	    System.exit(-1);
+	}
+	
+	if ( srcPort < 1024 || srcPort >= 65535 ) {
+	    System.err.println("Invalid port. (must be above 1024)");
+	    System.exit(-1);
+	}
+	
+	float canalloss = -1;
+	
+	try {
+	    canalloss = Integer.valueOf(args[4]);
+	} catch (NumberFormatException e) {
+	    System.err.println("Invalid network loss percentage. (must be a float in [0.0, 100.0[)");
+	    System.exit(-1);
+	}
+	
+	if ( canalloss < 0 || canalloss >= 100 ) {
+	    System.err.println("Invalid network loss percentage. (must be a float in [0.0, 100.0[)");
+	    System.exit(-1);
+	}
+	
+	int testParameter = Integer.valueOf(args[5]);
+	
+	EmetteurThreadMulti sendingThread = new EmetteurThreadMulti(sendIP, srcPort + 1);
+	
+	System.out.println("Listening for file " + fileName + " (" + fileSize + " bytes) at port " + srcPort + "\n");
+	/**
+	 * preparation for the decoding
+	 */
+	sendingThread.start();
+	// the RTT
+	
+	Treatment treatmentThread = new Treatment(Utils.RTT);
+	
+	/**
+	 * Get the first packet to get the size of the file
+	 */
+	MulticastSocket serverSocketForTheSize = null;
+	try {
+	    serverSocketForTheSize = new MulticastSocket(srcPort);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *serverSocketForTheSize*. The resource is referenced at lines 163, 179. The resource is closed at line 194. There are other execution paths that do not contain closure statements, for example, when DatagramSocket.receive throws an exception. Ensure *serverSocketForTheSize* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).